assembly_uploader/ena_queries.py

```diff
@@ -86,36 +53,134 @@ def parse_accession(accession):
 
 
 class EnaQuery:
-    def __init__(self, accession, username=None, password=None):
-        self.url = ENA_API_URL
+    def __init__(self, accession, private=False):
+        self.private_url = "https://www.ebi.ac.uk/ena/submit/report/"
+        self.public_url = "https://www.ebi.ac.uk/ena/portal/api/search"
         self.accession = accession
         self.acc_type = parse_accession(accession)
-        username = username or os.getenv("ENA_WEBIN")
-        password = password or os.getenv("ENA_WEBIN_PASSWORD")
+        username = os.getenv("ENA_WEBIN")
+        password = os.getenv("ENA_WEBIN_PASSWORD")
+        if username is None or password is None:
+            logging.error("ENA_WEBIN and ENA_WEBIN_PASSWORD are not set")
         if username and password:
             self.auth = (username, password)
         else:
             self.auth = None
-        self.data = build_data(self.accession, self.acc_type)
-
-    def post_request(self):
-        if self.auth:
-            response = requests.post(
-                self.url,
-                data=self.data,
-                auth=self.auth,
-                **get_default_connection_headers(),
+        self.private = private
+
+    def post_request(self, data):
+        response = requests.post(
+            self.public_url, data=data, **get_default_connection_headers()
+        )
+        return response
+
+    def get_request(self, url):
+        response = requests.get(url, auth=self.auth)
+        return response
+
+    def check_api_error(self, response):
+        try:
+            data = json.loads(response.text)[0]
+            return data
+        except NoDataException:
+            logging.error("Could not find {} in ENA".format(self.accession))
+        except (IndexError, TypeError, ValueError, KeyError):
+            logging.error(
+                "Failed to fetch {}, returned error: {}.".format(
+                    self.accession, response.text
+                )
             )
+
+    def get_study(self, attempt=0):
```

Comment:
I think this might be a bit more readable with something like
```python
class EnaQuery:
    def get_study(self, ...):
        if self.is_private:
            return self._get_private_study()
        else:
            return self._get_public_study()
    
    def _get_private_study(self, ...):
        ....
    
    def _get_public_study(self, ...):
        ....

    def _get_private_run(self, ...):
        ...
...
```
Just since the logic is so different for private/public, it might be a bit easy to read (and unit test) if they are just separate private methods?